src/Knp/Rad/ResourceResolver/Caster/VariableCaster.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Knp\Rad\ResourceResolver\Caster;
+
+use Knp\Rad\ResourceResolver\Caster;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+final class VariableCaster implements Caster
+{
+    /**
+     * @var RequestStack $requestStack
+     */
+    private $requestStack;
+
+    /**
+     * @param RequestStack $requestStack
+     */
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function cast($value)
+    {
+        $searched   = substr($value, 1);
+        $parameters = $this
+            ->requestStack
+            ->getCurrentRequest()
+            ->attributes
+            ->get('_route_params')
+        ;
+
+        foreach ($parameters as $parameter => $value) {
```

Comment:
haha I was joking about "is null". It's just you say (and repeat) `$parareters` instead of `$parameters`.

![](http://www.thebuzzmedia.com/wp-content/uploads/2011/05/troll-face-high-resolution.png)